usr/share/rear/format/USB/default/300_format_usb_disk.sh

```diff
@@ -133,43 +104,78 @@ else
         # so the number of the partition that can be set up next has to be one more (i.e. now 2):
         current_partition_number=$(( current_partition_number + 1 ))
     fi
+fi
 
-    if is_positive_integer $USB_BOOT_PART_SIZE ; then
-        # Create a boot partition for the bootloader config/plugins/modules, the kernel and the ReaR recovery system initrd.
-        # Round boot partition size to nearest block size to make the next partition (the data partition) also align to the block size:
-        USB_BOOT_PART_SIZE=$(( ( USB_BOOT_PART_SIZE + ( USB_PARTITION_ALIGN_BLOCK_SIZE / 2 ) ) / USB_PARTITION_ALIGN_BLOCK_SIZE * USB_PARTITION_ALIGN_BLOCK_SIZE ))
-        LogPrint "Creating boot partition $RAW_USB_DEVICE$current_partition_number with size $USB_BOOT_PART_SIZE MiB aligned at $USB_PARTITION_ALIGN_BLOCK_SIZE MiB"
-        # Calculate byte values:
-        local boot_partition_size_bytes=$(( USB_BOOT_PART_SIZE * MiB_bytes ))
-        # The end byte is the last byte that belongs to that partition so that one must be careful to use "start_byte + partition_size_in_bytes - 1":
-        local boot_partition_end_byte=$(( boot_partition_start_byte + boot_partition_size_bytes - 1 ))
-        if ! parted -s $RAW_USB_DEVICE unit B mkpart primary $boot_partition_start_byte $boot_partition_end_byte ; then
-            Error "Failed to create boot partition $RAW_USB_DEVICE$current_partition_number"
-        fi
-        # Set the right flag for the boot partition unless no flag should be set:
-        if ! is_false $boot_partition_flag ; then
-            LogPrint "Setting '$boot_partition_flag' flag on boot partition $RAW_USB_DEVICE$current_partition_number"
-            if ! parted -s $RAW_USB_DEVICE set $current_partition_number $boot_partition_flag on ; then
-                Error "Failed to set '$boot_partition_flag' flag on boot partition $RAW_USB_DEVICE$current_partition_number"
-            fi
-            # When the flag was set for the boot partition do not also set this flag for the data partition below:
-            boot_partition_flag="false"
+# In case of EFI boot we need a EFI system partition
+if is_true "$FORMAT_EFI" ; then
+    LogPrint "The --efi toggle was used with format - making an EFI bootable device $RAW_USB_DEVICE"
```

Comment:
You may leave out "The --efi toggle was used with format - " and only show
"Making an EFI bootable device $RAW_USB_DEVICE"
because plain `rear format /dev/sdX` (i.e. without any --efi toggle)
also makes it EFI bootable.